src/plat/spike/machine/hardware.c

```diff
@@ -140,24 +140,35 @@ ackInterrupt(irq_t irq)
 }
 
 static inline uint64_t get_cycles(void)
-#if __riscv_xlen == 32
 {
+    UNUSED unsigned long mtvec;
+
+    if (config_set(CONFIG_SEL4_RV_MACHINE)) {
+        /* Trap to riscv-pk to get time */
+        mtvec = read_csr_env(tvec);
+        write_csr(mtvec, 0x80000004ul);
+    }
+#ifdef CONFIG_ARCH_RISCV_RV32
     uint32_t nH, nL;
     __asm__ __volatile__ (
         "rdtimeh %0\n"
         "rdtime  %1\n"
         : "=r" (nH), "=r" (nL));
+    if (config_set(CONFIG_SEL4_RV_MACHINE)) {
+        write_csr(mtvec, mtvec);
+    }
```

Comment:
I would suggest having the rv32 and rv64 arms of this code both produce a return value and then have a single rest of mtvec at the end of the function prior to the return